requests/adapters.py

```diff
@@ -299,8 +300,11 @@ def get_connection(self, url, proxies=None):
         proxy = select_proxy(url, proxies)
 
         if proxy:
-            proxy = prepend_scheme_if_needed(proxy, 'http')
-            proxy_manager = self.proxy_manager_for(proxy)
+            _proxy = prepend_scheme_if_needed(proxy, 'http')
+            proxy_url = parse_url(_proxy)
+            if proxy_url.host == None:
+                raise(ProxyError("URL for proxy is invalid: {}".format(proxy)))
```

Comment:
Let's make this error message a bit clearer and say why the URL is invalid.

Reply:
You also don't need to wrap the Exception instance in parens after `raise`.

i.e.
```python
raise ProxyError("URL for proxy is invalid: {}".format(proxy))
```

Reply:
To gauge the exact reason is a bit tricky. Because, several things can trigger this code path. It's not always a missing host. It could also just be typos like:
```
- missing '/' in http://
- additional '/' after http://
```

Reply:
So I get that it isn't completely clear what's causing this error but *something* in the URL will be causing our parser to think the host doesn't exist. Steering the user in that direction seems better than an ambiguous invalidity.